compiler/rustc_typeck/src/check/demand.rs

```diff
@@ -356,9 +356,36 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 })
                 .collect();
 
-            if let [variant] = &compatible_variants[..] {
+            if self.is_hir_id_from_struct_pattern_shorthand_field(expr.hir_id, expr.span) {
+                if let Ok(code) = self.tcx.sess.source_map().span_to_snippet(expr.span) {
+                    if let [variant] = &compatible_variants[..] {
+                        // Just a single matching variant.
+                        err.span_suggestion_verbose(
```

Comment:
I actually don't know the difference between verbose and non-verbose suggestions, but the suggestion was not rendering inline without verbose.

Reply:
That's the difference ;)

https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/diagnostic/struct.Diagnostic.html#method.span_suggestion_verbose

Reply:
Hm. Is there a heuristic for inline rendering suggestions not marked verbose? Because the multipart_suggestion I replaced with this was rendering fine.

Whatev I guess :stuck_out_tongue_closed_eyes:

Reply:
Yeah, there are heuristics, I forget what they are though. I think they're mainly based on the length of the message and suggested code.

Reply:
Could you turn this into a multipart suggestion and remove the source map snippet call? It'll improve the displayed suggestion diff.

Reply:
So I modeled this code off of another suggestion that catches the same case: https://github.com/rust-lang/rust/blob/master/compiler/rustc_typeck/src/check/demand.rs#L873

If I remove the snippet call, I still need to extract the variant name. I guess we can get that from hir.

If I do turn this into a multipart, should it render like:

```
Foo { bar: Some(bar) }
      ++++++++++   +
```

Reply:
Yep, that's how it should render.

> So I modeled this code off of another suggestion that catches the same case: https://github.com/rust-lang/rust/blob/master/compiler/rustc_typeck/src/check/demand.rs#L873

I'd welcome a PR that converts that one as well ;)

Generally multipart suggestions are preferred, as long as they don't make the code _too_ complicated.

Reply:
> If I remove the snippet call, I still need to extract the variant name. I guess we can get that from hir.

The snippet call doesn't have anything to do with the variant name, unless I'm misunderstand...? I don't think you'd have to change that part.

Reply:
whoopsy, I meant the field name. Whatev, I got a change to make this back into a multipart. Gimme a sec.